compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -2840,13 +2842,18 @@ public void visit(BLangWhile whileNode) {
     }
 
     public void visit(BLangLock lockNode) {
-        if (lockNode.lockVariables.isEmpty()) {
+        if (lockNode.lockVariables.isEmpty() && lockNode.fieldVariables.isEmpty()) {
             this.genNode(lockNode.body, this.env);
             return;
         }
+
+        //remove objects initialized within lock
+        lockNode.fieldVariables.keySet().removeIf(symbol -> symbol.varIndex == null);
```

Comment:
shall we check this a bit? not sure checking null is ok

Reply:
I have checked by debugging and it does not cause a removal of any unintended variables other than the variables defined within a lock. But, I agree we should have a better way to identify this.